dts/bindings/spi/nxp,s32-spi.yaml

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP S32 SPI controller
+
+compatible: "nxp,s32-spi"
+
+include: [spi-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  interrupts:
+    required: true
+
+  num-cs:
+    type: int
+    required: true
+    description: |
+      The number of the Chip Select signals.
+
+  clock-frequency:
+    type: int
+    required: true
+    description: |
+      Module clock frequency in Hz.
+
+  pinctrl-0:
```

Comment:
These should be in the pinctrl bindings file.

Reply:
indeed, but are not required by default, and without it is not possible to configure SPI pins in the shim driver 
https://github.com/zephyrproject-rtos/zephyr/blob/c80232f16d477906346cab12b60a17794148bcbc/dts/bindings/pinctrl/pinctrl-device.yaml#L13

Reply:
Including `pinctrl-device.yaml` should be sufficient as has been done by other driver bindings files. Kindly delete these properties.

Also I do not see any SPI pin defines in the board `pinctrl` file, are these not needed?

Reply:
Hi, as i see there are a lot of vendors are including this in their bindings (e.g Kinetis DSPI controller: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/spi/nxp%2Ckinetis-dspi.yaml#L38).

For the second question, there is a SPI device on the board but there is no driver for it now and actually can not be fully tested, so i didn't configure the pins for it. SPI driver is tested by using the I/O headers (to loopback or transfer data between SPI instances on the board).